presto-cli/src/main/java/com/facebook/presto/cli/StatusPrinter.java

```diff
@@ -283,7 +283,9 @@ private void printQueryInfo(QueryResults results)
             }
 
             assert terminalWidth >= 75;
-            int progressWidth = (min(terminalWidth, 100) - 75) + 17; // progress bar is 17-42 characters wide
+            String formattedWallTime = formatTime(wallTime);
+            // progress bar is 17-42 characters wide when the wall time has 5 characters (hh:mm)
+            int progressWidth = (min(terminalWidth, 100) - 75) + 22 - formattedWallTime.length();
```

Comment:
@findepi  Revised the code to avoid use constant `22`. For 100, and 75, it's because `(min(terminalWidth, 100) - 75)` will be in [0, 25], thus the final results will be in [17, 17 + 25 = 42]